libraries/GSM/src/GSM3CircularBuffer.cpp

```diff
@@ -198,121 +198,121 @@ bool GSM3CircularBuffer::extractSubstring(const char* from, const char* to, char
 	byte h2;
 	byte b;
 	int i;
-	
+
 //DEBUG
 //Serial.println("Beginning extractSubstring");
 //Serial.print("head,tail=");Serial.print(int(head));Serial.print(",");Serial.println(int(tail));
-	
+
 	if(!locate(from, head, tail, 0, &t1))
 		return false;
-		
+
 //DEBUG
 //Serial.println("Located chain from.");
 
-	t1++; //To point the next.
+	++t1; //To point the next.
 	if(!locate(to, t1, tail, &h2, 0))
 		return false;
-		
-//DEBUG		
+
+//DEBUG
 //Serial.println("Located chain to.");
 /*Serial.print("t1=");Serial.println(int(t1));
 Serial.print("h2=");Serial.println(int(h2));*/
 
 
-	for(i=0,b=t1;i<bufsize, b!=((h2) & __BUFFERMASK__); i++, b=(b+1)& __BUFFERMASK__)
-		buffer[i]=theBuffer[b];
-	buffer[i]=0;
+	for(i = 0, b = t1; i < bufsize, b != ((h2) & __BUFFERMASK__); ++i, b = (b + 1) & __BUFFERMASK__)
+		buffer[i] = theBuffer[b];
+	buffer[i] = 0;
 
-//DEBUG		
+//DEBUG
 //Serial.println("");
-//Serial.println("Finishing extractSubstring");	
-	
-	return true;	
+//Serial.println("Finishing extractSubstring");
+
+	return true;
 }
 
 int GSM3CircularBuffer::readInt()
 {
-	int res=0;
-	byte c;
-	bool anyfound=false;
-	bool negative=false;
-	for(byte b=head + 1; b!=tail; b=(b+1)& __BUFFERMASK__)
+	int res = 0;
+	byte charakter;
+	bool anyfound = false;
+	bool negative = false;
+	for(byte index = head + 1; index != tail; index = (index + 1) & __BUFFERMASK__)
 	{
-		c=theBuffer[b];
-		if((c==' ' )&&(!anyfound))
+		charakter = theBuffer[index];
+		if((charakter == ' ' ) && (!anyfound))
 		{
-		} else if((c=='-' )&&(!anyfound))
+		} else if((charakter == '-' ) && (!anyfound))
 		{
-			negative=true;
-			anyfound=true;  // Don't admit blanks after -
-		} else if((c>='0')&&(c<='9'))
+			negative = true;
+			anyfound = true;  // Don't admit blanks after -
+		} else if((charakter >= '0') && (charakter <= '9'))
 		{
-			anyfound=true;
-			res=(res*10)+(int)c-48;
+			anyfound = true;
+			res = (res * 10) + static_cast<int>(charakter) - 48;
 		}
 		else
 			{
 				if(negative)
-					res=(-1)*res;
+					res = (-1) * res;
 				return res;
 			}
 	}
 	if(negative)
-		res=(-1)*res;
+		res = (-1) * res;
 	return res;
 }
 
 void GSM3CircularBuffer::debugBuffer()
 {
-	byte h1=head;
-	byte t1=tail;
+	byte h1 = head;
+	byte t1 = tail;
 	Serial.println();
 	Serial.print(h1);
 	Serial.print(" ");
 	Serial.print(t1);
 	Serial.print('>');
-	for(byte b=h1; b!=t1; b=(b+1)& __BUFFERMASK__)
+	for(byte b = h1; b != t1; b = (b + 1) & __BUFFERMASK__)
 		printCharDebug(theBuffer[b]);
 	Serial.println();
 }
 
-void GSM3CircularBuffer::printCharDebug(uint8_t c)
+void GSM3CircularBuffer::printCharDebug(const uint8_t character)
 {
-	if((c>31)&&(c<127))
-		Serial.print((char)c);
+	if((character > static_cast<uint8_t>(31)) && (character < static_cast<uint8_t>(127)))

```

Comment:
Okay, both variants result obviously in the same binary result. I expected more memory usage because 31 and 127 are integer literals not unsigned integer literals.